### PR TITLE
docs: update using components page

### PR DESCRIPTION
**Pull request by virkt25**
connected to #988 

Same as this PR https://github.com/strongloop/loopback.io/pull/639 but since we moved docs before it landed, re-creating this. 

## Checklist

- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: docs/site/Using-components.md [hunk 1]
```diff
@@ -7,24 +7,43 @@ sidebar: lb4_sidebar
...
+Components can contribute the following items:
+
+- [Controllers](Controllers.html)
+- Providers of additional [Context values](Context.html)
 
- - [Controllers](Controllers.md)
- - Providers of additional [Context values](Context.md)
+LoopBack 4 was built with extensibility in mind and this includes Components,
+which can be allowed to contribute additional artifacts by adding a Mixin
+to your Application class. This doesn't change how a a Component is registered
```
Comment (dhmlau) [inline docs/site/Using-components.md]: an extra `a` in `how a a Component`

File: docs/site/Using-components.md [hunk 2]
```diff
@@ -7,24 +7,43 @@ sidebar: lb4_sidebar
...
 
-In the future (before the GA release), components will be able to contribute additional items:
+- [Repositories](Repositories.html) can be contributed by a Component by adding
+  `RepositoryMixin` from `@loopback/repository` to your Application
+- [Booters](Booting-an-Application.html#booters) can be contributed by a Component by adding
+  `BootMixin` from `@loopback/boot` to your Application
 
- - Models
- - [Repositories](Repositories.md)
+**Note:** Always check a components instructions to see if it requires
```
Comment (dhmlau) [inline docs/site/Using-components.md]: either `a component's` or `a component`?

File: docs/site/Using-components.md [hunk 3]
```diff
@@ -7,24 +7,43 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/Using-components.html
 summary:
 ---
-Components are an important part of LoopBack 4. We are keeping the core small and easy to extend, making it easy for independent developers to contribute additional features to LoopBack. Components are the vehicle for bringing more goodness to your applications.
 
-A typical LoopBack component is an [npm](https://www.npmjs.com/) package exporting a Component class which can be added to your application.
+Components play an important part in the extensibility of LoopBack 4.
+A Component makes it easy for independent developers to contribute additional
+features to LoopBack. Components serve as a vehicle to group extension
+contributions such as Context Bindings and various Artifacts to allow easier
```
Comment (shimks) [inline docs/site/Using-components.md]: Link to Context.md here would be great

File: docs/site/Using-components.md [hunk 4]
```diff
@@ -7,24 +7,43 @@ sidebar: lb4_sidebar
...
+const app = new RestApplication();
+// Add component to Application, which provides bindings used to resolve
+// authenticated requests in a Sequence.
 app.component(AuthenticationComponent);
 ```
 
-In general, components can contribute the following items:
+Components can contribute the following items:
+
+- [Controllers](Controllers.html)
```
Comment (shimks) [inline docs/site/Using-components.md]: .md

File: docs/site/Using-components.md [hunk 5]
```diff
@@ -7,24 +7,43 @@ sidebar: lb4_sidebar
...
+- Providers of additional [Context values](Context.html)
```
Comment (shimks) [inline docs/site/Using-components.md]: .md

File: docs/site/Using-components.md [hunk 6]
```diff
@@ -7,24 +7,43 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/Using-components.html
 summary:
 ---
-Components are an important part of LoopBack 4. We are keeping the core small and easy to extend, making it easy for independent developers to contribute additional features to LoopBack. Components are the vehicle for bringing more goodness to your applications.
 
-A typical LoopBack component is an [npm](https://www.npmjs.com/) package exporting a Component class which can be added to your application.
+Components play an important part in the extensibility of LoopBack 4.
+A Component makes it easy for independent developers to contribute additional
+features to LoopBack. Components serve as a vehicle to group extension
+contributions such as [Context Bindings](Context.md) and various Artifacts to allow easier
```
Comment (raymondfeng) [inline docs/site/Using-components.md]: `Artifacts` should be `artifacts`.

File: docs/site/Using-components.md [hunk 7]
```diff
@@ -7,24 +7,43 @@ sidebar: lb4_sidebar
...
 
-```js
-import {Application} from '@loopback/core';
-import {AuthenticationComponent} from '@loopback/authentication';
+A typical LoopBack component is an [npm](https://www.npmjs.com) package
+exporting a Component class which can be added to your application.
 
-const app = new Application();
+```ts
+import { RestApplication } from "@loopback/rest";
```
Comment (raymondfeng) [inline docs/site/Using-components.md]: The formatting does not conform to our style. We don't use extra spaces for the `{}`.

**Pull request merged**